src/DripsDeployer.sol

```diff
@@ -0,0 +1,412 @@
+        BaseModule(dripsDeployer_, "NFTDriver")
+    {
+        // slither-disable-next-line too-many-digits
+        _deployProxy(proxyAdmin_, type(NFTDriver).creationCode);
+    }
+
+    function logicArgs() public view override returns (bytes memory) {
+        return abi.encode(_dripsModule().drips(), _callerModule().caller(), driverId);
+    }
+
+    function nftDriver() public view returns (NFTDriver) {
+        return NFTDriver(proxy());
+    }
+}
+
+contract ImmutableSplitsDriverModule is DriverModule(2) {
+    function args() public view override returns (bytes memory) {
+        return abi.encode(dripsDeployer, proxyAdmin);
+    }
+
+    constructor(address dripsDeployer_, address proxyAdmin_)
+        BaseModule(dripsDeployer_, "ImmutableSplitsDriver")
+    {
+        // slither-disable-next-line too-many-digits
+        _deployProxy(proxyAdmin_, type(ImmutableSplitsDriver).creationCode);
+    }
+
+    function logicArgs() public view override returns (bytes memory) {
+        return abi.encode(_dripsModule().drips(), driverId);
+    }
+
+    function immutableSplitsDriver() public view returns (ImmutableSplitsDriver) {
+        return ImmutableSplitsDriver(proxy());
+    }
+}
+
+contract RepoDriverModule is CallerDependentModule, DriverModule(3) {
+    OperatorInterface public immutable operator;
+    bytes32 public immutable jobId;
+    uint96 public immutable defaultFee;
+
+    function args() public view override returns (bytes memory) {
+        return abi.encode(dripsDeployer, proxyAdmin, operator, jobId, defaultFee);
+    }
+
+    constructor(
+        address dripsDeployer_,
+        address proxyAdmin_,
+        OperatorInterface operator_,
+        bytes32 jobId_,
+        uint96 defaultFee_
+    ) BaseModule(dripsDeployer_, "RepoDriver") {
+        operator = operator_;
+        jobId = jobId_;
+        defaultFee = defaultFee_;
+        // slither-disable-next-line too-many-digits
+        _deployProxy(proxyAdmin_, type(RepoDriver).creationCode);
+        repoDriver().initializeAnyApiOperator(operator, jobId, defaultFee);
+    }
+
+    function logicArgs() public view override returns (bytes memory) {
+        return abi.encode(_dripsModule().drips(), _callerModule().caller(), driverId);
+    }
+
+    function repoDriver() public view returns (RepoDriver) {
+        return RepoDriver(proxy());
+    }
+}
+
+/// @notice Creates a contract under a deterministic addresses
+/// derived only from the deployer's address and the salt.
+/// The deployment is a two-step process, first, a proxy is deployed using CREATE2 with
+/// the given salt, and then it's called with the bytecode of the deployed contract,
+/// the proxy uses it to deploy the contract using regular CREATE.
+/// If the deployed contract's constructor reverts, the proxy also reverts.
+/// If the proxy call has a non-zero value, it's passed to the deployed contract's constructor.
+/// Based on the bytecode from https://github.com/0xsequence/create3.
```

Comment:
I still need to look into it to understand the differences.

Reply:
It's the same. There's 1 implementation of `Create3` passed around that everybody adds to their libraries. I've just linked the (IIUC) the original one.

Reply:
If you want weird trivia and a piece of a lovely wasted time, I wrote a modified `Create3` that can be used directly by EOAs: https://gist.github.com/CodeSandwich/3022c63e3660a8f1b2dee2a235687a88 Regular `Create3` requires 2 transactions, first deploying the proxy and then calling it with the deployed init code. This is unusable for EOAs, because you may be front-run. The modified `Create3` stores its own deployer and lets only that address do the deployment, enabling bypassing the singleton factory and simplifying the address deriving chain. There's only one problem, `CREATE2` exists only as an opcode, you can't make a `CREATE2` transaction :rofl: